kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -699,147 +688,19 @@ def parse(s: str) -> list[T]:
-            kevm_cli_args.kprove_legacy_args,
-        ],
-    )
-    prove_legacy_args.add_argument(
-        '--bug-report-legacy', default=False, action='store_true', help='Generate a legacy bug report.'
-    )
-
-    command_parser.add_parser(
-        'view-kcfg',
-        help='Explore a given proof in the KCFG visualizer.',
-        parents=[kevm_cli_args.logging_args, kevm_cli_args.k_args, kevm_cli_args.spec_args],
-    )
-
-    command_parser.add_parser(
-        'show-kcfg',
-        help='Print the CFG for a given proof.',
-        parents=[
-            kevm_cli_args.logging_args,
-            kevm_cli_args.k_args,
-            kevm_cli_args.kcfg_show_args,
-            kevm_cli_args.spec_args,
-            kevm_cli_args.display_args,
-        ],
-    )
-
-    run_args = command_parser.add_parser(
-        'run',
-        help='Run KEVM test/simulation.',
-        parents=[
-            kevm_cli_args.logging_args,
-            kevm_cli_args.target_args,
-            kevm_cli_args.evm_chain_args,
-            kevm_cli_args.k_args,
-        ],
-    )
-    run_args.add_argument('input_file', type=file_path, help='Path to input file.')
-    run_args.add_argument(
-        '--output',
-        default=KRunOutput.PRETTY,
-        type=KRunOutput,
-        choices=list(KRunOutput),
-    )
-    run_args.add_argument(
-        '--expand-macros',
-        dest='expand_macros',
-        default=True,
-        action='store_true',
-        help='Expand macros on the input term before execution.',
-    )
-    run_args.add_argument(
-        '--no-expand-macros',
-        dest='expand_macros',
-        action='store_false',
-        help='Do not expand macros on the input term before execution.',
-    )
-    run_args.add_argument(
-        '--debugger',
-        dest='debugger',
-        action='store_true',
-        help='Run GDB debugger for execution.',
-    )
-
-    kast_args = command_parser.add_parser(
-        'kast',
-        help='Run KEVM program.',
-        parents=[
-            kevm_cli_args.logging_args,
-            kevm_cli_args.target_args,
-            kevm_cli_args.evm_chain_args,
-            kevm_cli_args.k_args,
-        ],
-    )
-    kast_args.add_argument('input_file', type=file_path, help='Path to input file.')
-    kast_args.add_argument(
-        '--output',
-        default=PrintOutput.KORE,
-        type=PrintOutput,
-        choices=list(PrintOutput),
-    )
+    command_parser = VersionOptions.parser(command_parser)
+    command_parser = KompileSpecOptions.parser(command_parser)
+    command_parser = ProveOptions.parser(command_parser)
+    command_parser = PruneProofOptions.parser(command_parser)
+    command_parser = ProveLegacyOptions.parser(command_parser)
+    command_parser = ViewKCFGOptions.parser(command_parser)
+    command_parser = ShowKCFGOptions.parser(command_parser)
+    command_parser = RunOptions.parser(command_parser)
+    command_parser = KastOptions.parser(command_parser)
```

Comment:
I guess I'm not 100% thrilled about this design (pass the parser around and have it be modified). I would like a design where it was more like:

```
command_parser = CommandParser([VersionOptions, KompileSpecOptions, ProveOptions, ...])
```

for example. Or even have a super class `KEVMCLI`, which inherits alll the other options classes and puts them togethr in the way that's needed.

Reply:
I've implemented a change to the `pyk` version here: https://github.com/runtimeverification/pyk/pull/916/commits/9f6574681d58d787c63ee484cbf02b26410096ad similar to what is suggested here which should handle the dispatch issue, and having to manually call `parser` for each subcommand, but it's still passing around the parser. I'll see if I can eliminate that as well.